modules/mongodb/mongodb.go

```diff
@@ -96,6 +103,24 @@ func (m *Mongo) Collect() map[string]int64 {
 		m.Errorf("couldn't collecting dbstats metrics: %s", err)
 	}
 
+	if m.mongoCollector.isReplicaSet() {
+		// if we have replicate set based on the serverStatus response
+		// we add once the charts during runtime
+		m.addReplChartsOnce.Do(func() {
+			replCharts := replCharts.Copy()
+			for _, chart := range *replCharts {
+				err := m.charts.Add(chart)
+				if err != nil {
+					m.Errorf("failed to add replica set chart: %s", chart.ID)
+				}
+			}
+		})
```

Comment:
```suggestion
		m.addReplChartsOnce.Do(func() {
			if err := m.charts.Add(*replCharts.Copy()...); err != nil {
				m.Errorf("failed to add replica set chart: %v", err)
			}
		})
```

Why don't add all the charts at once?

Reply:
good point